### DISCUSSION TITLE
Gremlin.version()

### DISCUSSION THREAD

**Discussion by jgdervil** [JanusGraph/janusgraph#3004] [General]
What version of the Gremlin Server comes with JanusGraph v0.6.1 and  Is it possible to change this version?

**Comment by vtslab** [comment]
The Gremlin Server version for JanuGraph v0.6.1 is tinkerpop-3.5.1. You can find this from:  

- [https://github.com/JanusGraph/janusgraph/blob/v0.6.1/pom.xml](https://github.com/JanusGraph/janusgraph/blob/v0.6.1/pom.xml) 
- the ./lib folder of the janusgraph distribution

Practically this cannot be changed. You can try and do a custom build of JanusGraph but you will find that changing the TinkerPop version will result in all kinds of library version conflicts that are time-consuming to solve (thanks JG team, for doing this chore!)

**Reply by jgdervil** [reply]
Thanks @vtslab! I'm a newbie to this, and the reason I asked is because I an connecting to JanusGraph 0.6.1 using version 3.5.2 of the gremlin console. When I check the gremlin version it shows 1.21...what am I missing here?

**Comment by vtslab** [comment]
You can simply use the gremlin console that is shipped with JanusGraph:

$ cd janusgraph-full-0.6.1    

$ bin/gremlin.sh

The Gremlin.version() is the version of the Gremlin language and is independent of the TinkerPop version.

**Reply by jgdervil** [reply]
Ah ok thanks for the clarification @vtslab , is it possible that the 0.6.1 version of JanusGraph is using an older gremlin language version than the 0.5.3 version of JanusGraph?

**Reply by jgdervil** [reply]
Sorry for all the questions @vtslab , but you mentioned  janusgraph-full-0.6.1, is the this a zip file or directory in this repo? I tried to download it from the 0.6.1 tag in this repo but couldn't find the directory path you mentioned.

**Reply by vtslab** [reply]
Download links for the zip distributions and the signatures are to be found at:
[https://github.com/JanusGraph/janusgraph/releases](https://github.com/JanusGraph/janusgraph/releases)

**Reply by jgdervil** [reply]
thanks @vtslab! one last question...any idea why version 0.6.1 of JanusGraph would be running an older version of the Gremlin Language than JanusGraph version 0.5.3? I have been using the JanusGraph Docker project for my containerized deployment of v0.6.1: https://github.com/JanusGraph/janusgraph-docker where previously I was manually containerizing v0.5.3 using the .zip distribution.

**Reply by vtslab** [reply]
This turns out to be a good catch actually, because it is a ... bug. When using bin/gremlin.sh of the TinkerPop-3.5.1 disitribution, Gremlin.version() returns 3.5.1, for sure, as was your initial expectation. Sorry for not checking my answer more thoroughly.    
Somehow, the Gremlin Console of the JanusGraph distribution returns the version field of the bigtable-hbase-shaded jar's Manifest, instead that of the gremlin-core jar. It is not even clear to me whether this is a bug in TinkerPop or JanusGraph.

**Reply by li-boxuan** [reply]
JG issue: https://github.com/JanusGraph/janusgraph/issues/3027
TinkerPop issue: https://issues.apache.org/jira/browse/TINKERPOP-2761
